server/src/main/java/org/apache/druid/metadata/PendingSegmentRecord.java

```diff
@@ -0,0 +1,141 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.metadata;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.hash.Hasher;
+import com.google.common.hash.Hashing;
+import com.google.common.io.BaseEncoding;
+import org.apache.druid.java.util.common.StringUtils;
+import org.apache.druid.segment.realtime.appenderator.SegmentIdWithShardSpec;
+import org.joda.time.Interval;
+
+import javax.annotation.Nullable;
+import java.sql.ResultSet;
+
+/**
+ * Representation of a record in the pending segments table. <br/>
+ * Mapping of column in table to field:
+ * <li/>  id -> id (Unique identifier for pending segment)
+ * <li/>  sequence_name -> sequenceName (sequence name used for segment allocation)
+ * <li/>  sequence_prev_id -> sequencePrevId (previous segment id used for segment allocation)
+ * <li/>  upgraded_from_segment_id -> upgradedFromSegmentId (Id of the root segment from which this was upgraded)
+ * <li/>  task_allocator_id -> taskAllocatorId (Associates a task / task group / replica group with the pending segment)
+ */
+public class PendingSegmentRecord
+{
+  private final SegmentIdWithShardSpec id;
+  private final String sequenceName;
+  private final String sequencePrevId;
+  private final String upgradedFromSegmentId;
+  private final String taskAllocatorId;
+
+  public PendingSegmentRecord(
+      SegmentIdWithShardSpec id,
+      String sequenceName,
+      String sequencePrevId,
+      @Nullable String upgradedFromSegmentId,
+      @Nullable String taskAllocatorId
+  )
+  {
+    this.id = id;
+    this.sequenceName = sequenceName;
+    this.sequencePrevId = sequencePrevId;
+    this.upgradedFromSegmentId = upgradedFromSegmentId;
+    this.taskAllocatorId = taskAllocatorId;
+  }
+
+  public SegmentIdWithShardSpec getId()
+  {
+    return id;
+  }
+
+  public String getSequenceName()
+  {
+    return sequenceName;
+  }
+
+  public String getSequencePrevId()
+  {
+    return sequencePrevId;
+  }
+
+  /**
+   * The original pending segment using which this upgraded segment was created.
+   * Corresponds to the column upgraded_from_segment_id in druid_pendingSegments.
+   * Can be null for pending segments allocated before this column was added or for segments that have not been upgraded.
+   */
+  @Nullable
+  public String getUpgradedFromSegmentId()
+  {
+    return upgradedFromSegmentId;
+  }
+
+  /**
+   * task / taskGroup / replica group of task that allocated this segment.
+   * Corresponds to the column task_allocator_id in druid_pendingSegments.
```

Comment:
This line is not needed. It is self explanatory.
```suggestion
```

Reply:
Removed